src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -443,5 +485,62 @@ private static byte GetSeverityNumber(LogLevel logLevel)
                 return 1;
         }
     }
+
+    private static int SanitizeCategoryName(byte[] buffer, int cursor, ref int validNameLength, string categoryName)
+    {
+        int cursorStartIdx = cursor;
+
+        const int maxStr8LengthInBytes = (1 << 8) - 1;
+        if (categoryName.Length > maxStr8LengthInBytes)
+        {
+            // The size of categoryName should not be greater than maxStr8LengthInBytes.
+            return cursor;
```

Comment:
Should you be returning here? Do we not have to at least get the first 50 characters?